### ISSUE TITLE
RFC about (optionally) using bgpq3 to fetch data

### ISSUE THREAD

**Issue by bbonev** [6connect/irrpt#68]
I have noticed that certain as-set objects that include many ASNs and as-sets are quite slow to process or even yield memory error.

While running couple of simple tests with bgpq3, its performance is magnitudes faster.

What about providing a patch that detects if bgpq3 is present and using it instead of the built-in fetcher?

**Comment by nistorj** [comment]
@bbonev you are correct, bgpq3 does process the information today a whole lot quicker, I think it may have to do with the way php handles streams of data, will look into this further.

**Comment by bbonev** [comment]
Nope, I did some tests and the difference comes from parallel execution in bgpq3. Implementing parallelism is not impossible with php sockets, but I think that using bgpq3 as a back-end fetcher would be better.

I will prepare a PR as time permits
